src/map/packet_system.cpp

```diff
@@ -1611,7 +1611,7 @@ void SmallPacket0x04D(map_session_data_t* session, CCharEntity* PChar, CBasicPac
         return;
     }
 
-    if (PChar->m_GMlevel == 0 && !PChar->loc.zone->CanUseMisc(MISC_AH) && !PChar->loc.zone->CanUseMisc(MISC_MOGMENU))
+    if (!zoneutils::IsResidentialArea(PChar) && PChar->m_GMlevel == 0 && !PChar->loc.zone->CanUseMisc(MISC_AH) && !PChar->loc.zone->CanUseMisc(MISC_MOGMENU))
```

Comment:
is this intended to fix pulling up menu inside town but outside of mog house? ignore previous message, I'm still waking up I guess *yawns*

Reply:
Actually the opposite, inside town and outside of moghouse is already blocked, but you're currently being blocked if you enter your moghouse from a zone that doesn't have these things, such as Port Bastok. I guess the moghouse is part of the last zone you were in? Not really sure how that all works, but you seem to use the misc flags of whatever zone you entered from.

Reply:
yep. the mog house is part of the zone you entered from. I thought the town zones were flagged to allow and not block for this very reason.